### PR TITLE
add OpenAPI spec

### PR DESCRIPTION
**Pull request by js-kyle**
https://petstore.swagger.io/?url=https://raw.githubusercontent.com/UniversalMediaServer/api/openapi/openapi.yaml

### CODE REVIEW

**Comment by js-kyle** [comment]
Ready to have a look at it - ill do some final enhancements

**Comment by SubJunk** [comment]
Looks great, lots of detail in here

**Comment by SubJunk** [comment]
@js-kyle Is this ready for final review?

**Comment by js-kyle** [comment]
@SubJunk yep - we could add in some API contract testing against it at a later point

File: sequence.puml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+== Retrieve an existing metadata record ==
+
+API -> "API MongoDB": (2) Query for existing metadata record
+"API MongoDB" -> API: Existing metadata found
+API -> "Universal Media Server Client": If found, return API response
+
+== Create a new metadata record ==
+
+API -> IMDB : Query source API 1
```
Comment (SubJunk) [inline sequence.puml:16]:
```suggestion
API -> TMDB : Query source API 1
```

File: sequence.puml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+participant "Universal Media Server Client"
+participant "API"
+database "API MongoDB"
+
+"Universal Media Server Client" -> API: (1) UMS client requests metadata about a media file
+
+== Retrieve an existing metadata record ==
+
+API -> "API MongoDB": (2) Query for existing metadata record
+"API MongoDB" -> API: Existing metadata found
```
Comment (SubJunk) [inline sequence.puml:11]:
```suggestion
API -> "API MongoDB": Query for existing failed lookup record
"API MongoDB" -> API: Existing metadata found
```

File: sequence.puml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@startuml
+participant "Universal Media Server Client"
+participant "API"
+database "API MongoDB"
+
+"Universal Media Server Client" -> API: (1) UMS client requests metadata about a media file
```
Comment (SubJunk) [inline sequence.puml:6]: What are these numbers in brackets for? Can we use autonumber to make that easier to maintain?

**Pull request merged**